### PR TITLE
[SPARK-16019][yarn] Use separate RM poll interval when starting client AM.

### PR DESCRIPTION
**Pull request by vanzin**
Currently the code monitoring the launch of the client AM uses the value of
spark.yarn.report.interval as the interval for polling the RM; if someone
has that value to a really large interval, it would take that long to detect
that the client AM has started, which is not expected.

Instead, have a separate config for the interval to use when the client AM is
starting. The other config is still used in cluster mode, and to detect the
status of the client AM after it is already running.

Tested by running client and cluster mode apps with a modified value of
spark.yarn.report.interval, verifying client AM launch is detected before
that interval elapses.

### CODE REVIEW

**Comment by vanzin** [comment]
@tgravescs

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -991,8 +991,9 @@ private[spark] class Client(
   def monitorApplication(
       appId: ApplicationId,
       returnOnRunning: Boolean = false,
-      logApplicationReport: Boolean = true): (YarnApplicationState, FinalApplicationStatus) = {
-    val interval = sparkConf.get(REPORT_INTERVAL)
+      logApplicationReport: Boolean = true,
+      interval: Long = sparkConf.get(REPORT_INTERVAL)):
```
Comment (tgravescs) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: add new param to method description

**Review by tgravescs — COMMENTED** [review]
minor doc update, otherwise +1

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala [hunk 2]
```diff
@@ -136,10 +136,16 @@ package object config {
     .createWithDefault(true)
 
   private[spark] val REPORT_INTERVAL = ConfigBuilder("spark.yarn.report.interval")
-    .doc("Interval between reports of the current app status in cluster mode.")
+    .doc("Interval between reports of the current app status.")
     .timeConf(TimeUnit.MILLISECONDS)
     .createWithDefaultString("1s")
 
+  private[spark] val CLIENT_LAUNCH_MONITOR_INTERVAL =
+    ConfigBuilder("spark.yarn.am.launchMonitorInterval")
```
Comment (tgravescs) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala]: sorry missed this in my first pass. One thing here is that normally the spark.yarn.am.  configs are configs that apply to the am, this one is slightly different in that its how often the client pulls the am for status (applies more to the client). We aren't documenting it anyway but perhaps we should name something like spark.yarn.clientAMLaunchMonitorInterval?

**Comment by tgravescs** [comment]
+1, feel free to commit

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**